master/buildbot/util/service.py

```diff
@@ -470,15 +473,16 @@ def reconfigServiceWithBuildbotConfig(self, new_config):
         for n in old_set & new_set:
             old = old_by_name[n]
             new = new_by_name[n]
-            # detect changed class name
-            if reflect.qual(old.__class__) != reflect.qual(new.__class__):
-                removed_names.add(n)
-                added_names.add(n)
             # compare using ComparableMixin if they don't support reconfig
-            elif not hasattr(old, 'reconfigServiceWithBuildbotConfig'):
-                if not util.ComparableMixin.isEquivalent(old, new):
+            if not hasattr(old, 'reconfigServiceWithBuildbotConfig'):
```

Comment:
why do you change the logic order, could'n you just extract the reflect test into a method, and keep the same if order?

Reply:
I was going with the assumption that if a class doesn't have the `reconfigServiceWithBuildbotConfig` function, then it wouldn't have the new `canReconfigWithSibling` function.

I don't really understand when a class/instance could be missing `reconfigServiceWithBuildbotConfig`, so I just went with a guess.

Reply:
ah.. you are right... 
This might be some transition code. I think you may just drop the test, and we will see with test failures if there are still some services that are not BuildbotService.

Reply:
👍  I changed the code to only do the call to `canReconfigWithSibling` now.